### PR TITLE
Implement name validation before server calls on Org creation

### PR DESCRIPTION
**Pull request by Siddhant-K-code**
## Description

This PR addresses the need for adding a basic organization name validation before making server calls during the organization creation process.

### Impact

The goal is to optimize server load and elevate user experience by confirming that only valid organization names are processed. Currently, calls are made to the createTeam endpoint even when the name is empty. This approach allows users to promptly receive basic error messages for issues like empty fields and character limits, instead of sending a request to the server and then displaying the error on the UI, a process that typically takes approximately 1.2 seconds for updates.

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at b21dc8e</samp>

Added placeholders and validation for team and project names. This enhances the usability and feedback of the dashboard UI.

</details>

## Related Issue(s)

Fixes #

## How to test

- Open preview
- Create a new organization
    - Hit `Create organization` when input box is empty & see error message 👀
    - Now, do same with Org name of characters less than 3 and more than 64

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - sid-name-va7d9ee39a0</li>
	<li><b>🔗 URL</b> - <a href="https://sid-name-va7d9ee39a0.preview.gitpod-dev.com/workspaces" target="_blank">sid-name-va7d9ee39a0.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - sid-name-validation-org-creation-gha.17755</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-sid-name-va7d9ee39a0%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/projects/ProjectSettings.tsx [hunk 1]
```diff
@@ -191,6 +191,7 @@ export default function ProjectSettingsView() {
                     value={projectName}
                     error={badProjectName}
                     onChange={setProjectName}
+                    placeholder="My Project"
```
Comment (svenefftinge) [inline components/dashboard/src/projects/ProjectSettings.tsx:194]: Did you see a case where the name was empty?

File: components/dashboard/src/teams/NewTeam.tsx [hunk 2]
```diff
@@ -23,6 +23,22 @@ export default function NewTeamPage() {
     const createTeam = async (event: FormEvent) => {
         event.preventDefault();
 
+        /* Basic name validation before sending to the server to save time & API calls */
```
Comment (svenefftinge) [inline components/dashboard/src/teams/NewTeam.tsx:26]: It's fast enough and not worth maintaining validation logic twice.

**Review by svenefftinge — CHANGES_REQUESTED** [review]
Thanks @Siddhant-K-code for caring about the UX! 
We want to avoid maintaining validation logic and error messages twice, unless there is a  good reason. In this case, the UI feedback seems absolutely fine as it is. So I'd rather not add code unnecessarily. Let me know if I have missed something. 🙏

**Comment by Siddhant-K-code** [comment]
Absolutely, I'll proceed to close the PR. The main goal was to initiate communication with the server only after conducting fundamental validation on the frontend. However, I understand your concern regarding the management of duplicates. The primary reference for this was:

https://github.com/gitpod-io/gitpod/blob/1b68720f1c1a2bc26126a40dac05f34d2fa88f48/components/dashboard/src/projects/ProjectSettings.tsx#L51-L55

**Comment by svenefftinge** [comment]
Thanks! We should definitely be more consistent and rely on server-side validation as much as reasonable.

**Pull request closed without merging**